apollo-backend/src/main/java/io/logz/apollo/blockers/types/SingleRegionBlocker.java

```diff
@@ -0,0 +1,79 @@
+package io.logz.apollo.blockers.types;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.logz.apollo.blockers.BlockerInjectableCommons;
+import io.logz.apollo.blockers.BlockerType;
+import io.logz.apollo.blockers.BlockerTypeName;
+import io.logz.apollo.blockers.DeploymentBlockerFunction;
+import io.logz.apollo.blockers.RequestBlockerFunction;
+import io.logz.apollo.blockers.SingleRegionBlockerResponse;
+import io.logz.apollo.blockers.SingleRegionBlockerResponse.BlockerCause;
+import io.logz.apollo.models.Deployment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@BlockerType(name = BlockerTypeName.SINGLE_REGION)
+public class SingleRegionBlocker implements RequestBlockerFunction, DeploymentBlockerFunction {
+    private static final Logger logger = LoggerFactory.getLogger(SingleRegionBlocker.class);
+    private SingleRegionBlockerConfiguration singleRegionBlockerConfiguration;
+
+    @Override
+    public void init(String jsonConfiguration) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        singleRegionBlockerConfiguration = mapper.readValue(jsonConfiguration, SingleRegionBlockerConfiguration.class);
+    }
+
+    @Override
+    public boolean shouldBlock(BlockerInjectableCommons blockerInjectableCommons, Deployment deployment) {
+        List<Integer> runningServices = getAllOngoingDeploymentsByServiceId(deployment.getServiceId(), blockerInjectableCommons);
+        if (!runningServices.isEmpty()) {
+            logger.warn("There are already running deployments for the service {}", deployment.getServiceId());
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public SingleRegionBlockerResponse shouldBlock(List<Integer> serviceIds, int numOfEnvironments) {
+        List<Integer> serviceIdsToCheck = getServicesWithinSingleRegionBlocker(serviceIds);
+        if (!serviceIdsToCheck.isEmpty()) {
+            if (numOfEnvironments > 1) {
+                return new SingleRegionBlockerResponse(true, serviceIdsToCheck, BlockerCause.MULTIPLE_ENVIRONMENTS);
+            }
+        }
+        return new SingleRegionBlockerResponse(false);
+    }
+
+    private List<Integer> getServicesWithinSingleRegionBlocker(List<Integer> serviceIds) {
+        return serviceIds.stream()
+                .distinct()
+                .filter(singleRegionBlockerConfiguration.getServiceIds()::contains)
+                .collect(Collectors.toList());
+    }
+
+    private List<Integer> getAllOngoingDeploymentsByServiceId(int serviceId, BlockerInjectableCommons blockerInjectableCommons) {
+        return blockerInjectableCommons.getDeploymentDao().getAllOngoingDeploymentsByServiceId(serviceId).stream()
+                .map(deployment -> deployment.getServiceId())
```

Comment:
why do you need to convert to serviceIds? 
It doesn't matter if it's a list of ids or a list of Deployments if all you want to do is to check for .isEmpty()

Reply:
@matvey-mtn I am not convert a list of deployments into list of Integers, it is how it is comes from the Controllers.

Reply:
I think you didn't understand my question.
why do we need line 61 (`.map(deployment -> deployment.getServiceId())`) ?
we can just return 
```
return blockerInjectableCommons.getDeploymentDao().getAllOngoingDeploymentsByServiceId(serviceId)
```

Reply:
@maorshap 👆

Reply:
ah right, I confused with the second shouldBlock method. fix it.